### PR TITLE
Initialization for Crib

### PR DESCRIPTION
**Pull request by 0xAustinWang**
https://smartcontract-it.atlassian.net/browse/CCIP-4107

### CODE REVIEW

File: deployment/ccip/changeset/cs_home_chain.go [hunk 1]
```diff
@@ -82,9 +82,9 @@ func (c DeployHomeChainConfig) Validate() error {
 	return nil
 }
 
-// deployCapReg deploys the CapabilitiesRegistry contract if it is not already deployed
+// DeployCapReg deploys the CapabilitiesRegistry contract if it is not already deployed
 // and returns a deployment.ContractDeploy struct with the address and contract instance.
-func deployCapReg(
+func DeployCapReg(
```
Comment (AnieeG) [inline deployment/ccip/changeset/cs_home_chain.go]: These deploy funcs are intentionally made unexported. If you are using this please use the changeset corresponding to it. These should not be exported

**Review by scheibinger — DISMISSED** [review]
Tested in crib staging env and that fully worked 🎉

File: deployment/environment/devenv/don.go [hunk 2]
```diff
@@ -239,35 +239,51 @@ func (n *Node) CreateCCIPOCRSupportedChains(ctx context.Context, chains []JDChai
...
-			AccountAddr:      account,
-			AdminAddr:        n.adminAddr,
-			Ocr2Enabled:      true,
-			Ocr2IsBootstrap:  isBootstrap,
-			Ocr2Multiaddr:    n.multiAddr,
-			Ocr2P2PPeerID:    value(peerID),
-			Ocr2KeyBundleID:  ocr2BundleId,
-			Ocr2Plugins:      `{"commit":true,"execute":true,"median":false,"mercury":false}`,
+
+		// retry twice with 5 seconds interval to create JobDistributorChainConfig
```
Comment (b-gopalswami) [inline deployment/environment/devenv/don.go:243]: nit: update comment per retry interval

**Review by b-gopalswami — DISMISSED** [review]

File: deployment/environment/devenv/don.go [hunk 3]
```diff
@@ -377,6 +393,13 @@ func (n *Node) CreateJobDistributor(ctx context.Context, jd JobDistributor) (str
 		return "", err
 	}
 	// create the job distributor in the node with the csa key
+	resp, err := n.gqlClient.ListJobDistributors(ctx)
+	if err != nil {
+		return "", fmt.Errorf("Could not list job distrubutors: %w", err)
+	}
+	if len(resp.FeedsManagers.Results) > 0 {
+		return resp.FeedsManagers.Results[0].FeedsManagerParts.GetId(), nil
```
Comment (AnieeG) [inline deployment/environment/devenv/don.go]: I think this should be updated to check if the csakey matches with the JD, it should not just return the first one in the result

**Review by AnieeG — CHANGES_REQUESTED** [review]
Requesting change for the update of JD

**Review by AnieeG — APPROVED** [review]

**Review by b-gopalswami — APPROVED** [review]

**Review by 0xnogo — APPROVED** [review]

**Review by makramkd — APPROVED** [review]

**Pull request merged**